### PR TITLE
 Implement functionality for editing the user profile #66

### PR DESCRIPTION
**Pull request by fedyafedko**

### CODE REVIEW

File: StudyHub/StudyHub.BLL/Profiles/UserProfile.cs [hunk 1]
```diff
@@ -20,6 +20,9 @@ public UserProfile()
         CreateMap<GoogleJsonWebSignature.Payload, User>()
             .ForMember(dest => dest.UserName, opt => opt.MapFrom(src => src.Email));
 
+        CreateMap<User, StudentDTO>();
+
         CreateMap<User, UserDTO>();
```
Comment (Podgore) [inline StudyHub/StudyHub.BLL/Profiles/UserProfile.cs:25]: empty line

File: StudyHub/StudyHub.Common/DTO/UserInvitation/UpdateUserDTO.cs [hunk 2]
```diff
@@ -0,0 +1,9 @@
+﻿namespace StudyHub.Common.DTO.UserInvitation;
```
Comment (Podgore) [inline StudyHub/StudyHub.Common/DTO/UserInvitation/UpdateUserDTO.cs:1]: emtpy line

File: StudyHub/StudyHub/Validators/UserValidators/UpdateUserValidator.cs [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+using StudyHub.Common.DTO.UserInvitation;
+
+namespace StudyHub.Validators.UserValidator;
+
+public class UpdateUserValidator : AbstractValidator<UpdateUserDTO>
+{
+    public UpdateUserValidator()
+    {
+        RuleFor(x => x.FullName)
+            .NotEmpty();
```
Comment (Podgore) [inline StudyHub/StudyHub/Validators/UserValidators/UpdateUserValidator.cs:11]: add regex for validation

File: StudyHub/StudyHub/Validators/UserValidators/UpdateUserValidator.cs [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+        RuleFor(x => x.Telegram)
+            .NotEmpty();
```
Comment (Podgore) [inline StudyHub/StudyHub/Validators/UserValidators/UpdateUserValidator.cs:13]: also add some regex for validation

**Review by Podgore — CHANGES_REQUESTED** [review]

**Review by Podgore — DISMISSED** [review]

File: StudyHub/StudyHub/Controllers/UsersController.cs [hunk 5]
```diff
@@ -34,4 +36,11 @@ public async Task<IActionResult> SearchStudent([FromQuery] SearchRequest request
         var result = await _userService.GetStudents(request);
         return Ok(result);
     }
+
+    [HttpPut("[action]")]
```
Comment (aquaprogit) [inline StudyHub/StudyHub/Controllers/UsersController.cs:40]:
```suggestion
    [HttpPut("me")]
```

File: StudyHub/StudyHub/Controllers/UsersController.cs [hunk 6]
```diff
@@ -34,4 +36,11 @@ public async Task<IActionResult> SearchStudent([FromQuery] SearchRequest request
...
+    public async Task<IActionResult> Edit(Guid userId, UpdateUserDTO dto)
+    {
+        var result = await _userService.EditAsync(userId, dto);
```
Comment (aquaprogit) [inline StudyHub/StudyHub/Controllers/UsersController.cs:43]: get userId from HttpContext

File: StudyHub/StudyHub.Common/Utility/ValidationRegexes.cs [hunk 7]
```diff
@@ -3,4 +3,5 @@ public static class ValidationRegexes
 {
     public const string FullNameRegex = @"^[A-Za-z\s]*$";
     public const string PasswordRegex = @"^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)(?=.*[#$^+=!*()@%&]).{8,}$";
+    public const string Telegram = @".*?\B@\w{5}.*";
```
Comment (aquaprogit) [inline StudyHub/StudyHub.Common/Utility/ValidationRegexes.cs:6]: check Telegram actual regex

**Review by aquaprogit — CHANGES_REQUESTED** [review]

File: StudyHub/StudyHub.Common/Utility/ValidationRegexes.cs [hunk 8]
```diff
@@ -3,4 +3,5 @@ public static class ValidationRegexes
 {
     public const string FullNameRegex = @"^[A-Za-z\s]*$";
     public const string PasswordRegex = @"^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)(?=.*[#$^+=!*()@%&]).{8,}$";
+    public const string Telegram = @"^@[a-zA-Z0-9_]+";
```
Comment (aquaprogit) [inline StudyHub/StudyHub.Common/Utility/ValidationRegexes.cs:6]: image
keep it like it was

**Review by aquaprogit — APPROVED** [review]

File: StudyHub/StudyHub.BLL/Services/Interfaces/IUserService.cs [hunk 9]
```diff
@@ -1,9 +1,12 @@
...
+using StudyHub.Common.DTO.User.Student;
+using StudyHub.Common.DTO.UserInvitation;
 using StudyHub.Common.Requests;
 
 namespace StudyHub.BLL.Services.Interfaces;
 
 public interface IUserService
 {
-    Task<PageList<UserDTO>> GetStudents(SearchRequest request);
+    Task<PageList<StudentDTO>> GetStudents(SearchRequest request);
```
Comment (aquaprogit) [inline StudyHub/StudyHub.BLL/Services/Interfaces/IUserService.cs:10]: Async

**Review by aquaprogit — APPROVED** [review]

**Review by Podgore — APPROVED** [review]

**Pull request merged**